pkg/checks/runtime/runtime.go

```diff
@@ -0,0 +1,99 @@
+// sparrow
+// (C) 2024, Deutsche Telekom IT GmbH
+//
+// Deutsche Telekom IT GmbH and all other contributors /
+// copyright owners license this file to you under the Apache
+// License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package runtime
+
+import (
+	"github.com/caas-team/sparrow/pkg/checks"
+	"github.com/caas-team/sparrow/pkg/checks/dns"
+	"github.com/caas-team/sparrow/pkg/checks/health"
+	"github.com/caas-team/sparrow/pkg/checks/latency"
+)
+
+// Config holds the runtime configuration
+// for the various checks
+// the sparrow supports
+type Config struct {
+	Health  *health.Config  `yaml:"health" json:"health"`
+	Latency *latency.Config `yaml:"latency" json:"latency"`
+	Dns     *dns.Config     `yaml:"dns" json:"dns"`
+}
+
+// Empty returns true if no checks are configured
+func (c Config) Empty() bool {
+	return c.size() == 0
+}
+
+// Iter returns configured checks in an iterable format
+func (c Config) Iter() []checks.Runtime {
+	var configs []checks.Runtime
+	if c.Health != nil {
+		configs = append(configs, c.Health)
+	}
+	if c.Latency != nil {
+		configs = append(configs, c.Latency)
+	}
+	if c.Dns != nil {
+		configs = append(configs, c.Dns)
+	}
+	return configs
+}
+
+// size returns the number of checks configured
+func (c Config) size() int {
+	size := 0
+	if c.Health != nil {
+		size++
+	}
+	if c.Latency != nil {
+		size++
+	}
+	if c.Dns != nil {
+		size++
+	}
+	return size
+}
+
+// HasHealthCheck returns true if the check has a health check configured
+func (c Config) HasHealthCheck() bool {
+	return c.Health != nil
+}
+
+// HasLatencyCheck returns true if the check has a latency check configured
+func (c Config) HasLatencyCheck() bool {
+	return c.Latency != nil
+}
+
+// HasDNSCheck returns true if the check has a dns check configured
+func (c Config) HasDNSCheck() bool {
+	return c.Dns != nil
+}
```

Comment:
Refactor this to call HasCheck() internally

Reply:
Don't you think that this causes an unneeded indirection? I'm happy that we have dedicated functions that answer the question "is the check configured?". The HasCheck is a helper function only needed for the ReconcileChecks... 

I'm also not happy with it's existence, but I wouldn't move the logic for each check (currently the same for each check) in that function, as it isn't its place IMO. Instead I would revisit the ReconcileChecks function in the future and see whether we really need the `HasCheck` function.